modules/ml/catboost-model-parser/src/main/java/org/apache/ignite/ml/catboost/CatboostClassificationModelParser.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.ml.catboost;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import ai.catboost.CatBoostError;
+import ai.catboost.CatBoostModel;
+import org.apache.ignite.ml.inference.parser.ModelParser;
+import org.apache.ignite.ml.math.primitives.vector.NamedVector;
+
+/**
+ * Catboost Classification model parser.
+ */
+public class CatboostClassificationModelParser implements
+    ModelParser<NamedVector, Double, CatboostClassificationModel> {
```

Comment:
Could you please explain, why did you use NamedVector with CatBoost models

Reply:
Catboost require right order of features. So I can fetch order `model.getFeatureNames()` and fill input vector `floatInput` using expected order.

```java
float[] floatInput = new float[input.size()];
int index = 0;
for (String key: model.getFeatureNames()) {
    floatInput[index] = (float) input.get(key);
    index++;
}

return model.predict(floatInput, model.getFeatureNames()).get(0, 0);
```

Reply:
Hi, second parameter in predict for catboost model, isn't that supposed to be categorical features? Could you please help me understand how are categorical features passed here?